NangPaGo-be/src/test/java/com/mars/NangPaGo/domain/recipe/service/RecipeLikeServiceTest.java

```diff
@@ -0,0 +1,93 @@
+package com.mars.NangPaGo.domain.recipe.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+import com.mars.NangPaGo.domain.recipe.dto.RecipeLikeRequestDto;
+import com.mars.NangPaGo.domain.recipe.dto.RecipeLikeResponseDto;
+import com.mars.NangPaGo.domain.recipe.entity.Recipe;
+import com.mars.NangPaGo.domain.recipe.entity.RecipeLike;
+import com.mars.NangPaGo.domain.recipe.repository.RecipeLikeRepository;
+import com.mars.NangPaGo.domain.recipe.repository.RecipeRepository;
+import com.mars.NangPaGo.domain.user.entity.User;
+import com.mars.NangPaGo.domain.user.repository.UserRepository;
+import java.util.ArrayList;
+import java.util.Optional;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.transaction.annotation.Transactional;
+
+@Transactional
+@ExtendWith(MockitoExtension.class)
+class RecipeLikeServiceTest {
+
+    @Mock
+    private RecipeLikeRepository recipeLikeRepository;
+    @Mock
+    private RecipeRepository recipeRepository;
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private RecipeLikeService recipeLikeService;
+
+    @DisplayName("유저 좋아요 클릭")
+    @Test
+    void toggleRecipeLike() {
+        // given
+        String email = "dummy@nangpago.com";
+        Long recipeId = 1L;
+
+        User user = User.builder()
+            .email(email)
+            .build();
+
+        Recipe recipe = new Recipe();
+
+        RecipeLikeRequestDto requestDto =
+            new RecipeLikeRequestDto(email, recipeId);
+
+        // mocking
+        when(userRepository.findByEmail(anyString())).thenReturn(Optional.of(user));
+        when(recipeRepository.findById(anyLong())).thenReturn(Optional.of(recipe));
+        when(recipeLikeRepository.findByUserAndRecipe(user, recipe)).thenReturn(Optional.empty());
+
+        // when
+        RecipeLikeResponseDto recipeLikeResponseDto = recipeLikeService.toggleRecipeLike(requestDto);
+
+        // then
+        assertThat(recipeLikeResponseDto).isNotNull()
+            .extracting("liked")
+            .isEqualTo(true);
+    }
+
+    @DisplayName("유저가 좋아요한 상태인지 체크")
+    @Test
+    void isLikedByUser() {
+        // given
+        String email = "dummy@nangpago.com";
+        long recipeId = 1L;
+
+        User user = User.builder()
+            .email(email)
+            .build();
+
+        Recipe recipe = new Recipe();
```

Comment:
지금은 `Recipe` 엔티티에 `@Builder`가 빠져있어서 이렇게 작성하신 것 같네요.
일부러 `@Builder`를 뺀건 아닌것 같으니, 차라리 이참에 `@Builder`를 추가해서 사용할 수 있도록 바꾸는게 더 좋을 것 같아요.

```suggestion
        Recipe recipe = Recipe.builder()
            .id(recipeId)
            .build();
```

Reply:
코드 리뷰 반영했습니다.

커밋을 안 나눠놓아 같이 포함된 링크 첨부하겠습니다.

[7d17bd7](https://github.com/MARS-LIKELION/NangPaGo/pull/71/commits/7d17bd7dc1ea33e930e727f8c41f639083f71fee)